components/sup/src/manager/service/config.rs

```diff
@@ -227,6 +234,18 @@ impl ServiceConfig {
         self.needs_write = false;
         Ok(should_restart)
     }
+
+    pub fn reload_package(&mut self, package: &PackageInstall) -> Result<()> {
+        self.config_root = match self.config_root_from_config {
+            Some(ref root) => root.clone(),
+            None => package.installed_path.clone(),
+        };
+        self.pkg = Pkg::new(package,
```

Comment:
I see here why you've opted to change the constructor now! What we really want to do is update our RuntimeConfig so when the supervisor restarts it uses it. Else, we'll update the Pkg struct in ServiceConfig but the supervisor itself won't use the new user/group.

If it happens to today, it's just a side effect of how the code was structured. This reload function needs to reload everything about the Service which is informed by the Package